package/rmt-cli_bash-completion.sh

```diff
@@ -0,0 +1,131 @@
+_rmt-cli_products()
+{
+	local current_word options flags
+
+	current_word=${COMP_WORDS[COMP_CWORD]}
+	options="list enable disable help"
+	flags="--all --csv --release-stage="
+
+	if [[ ${COMP_CWORD} == 2 ]] ; then
+		COMPREPLY=( $(compgen -W "${options}" -- ${current_word}) )
+	elif [[ ${COMP_WORDS[2]} == help && ${COMP_CWORD} == 3 ]] ; then
+		COMPREPLY=( $(compgen -W "${options/help/}" -- ${current_word}) )
+	elif [[ ${current_word} == -* && ${COMP_WORDS[*]} =~ enable ]] ; then
+		COMPREPLY=( $(compgen -W "--all-modules" -- ${current_word}) )
+	elif [[ ${current_word} == -* && ${COMP_WORDS[2]} =~ ^(list|ls)$ ]] ; then
+		COMPREPLY=( $(compgen -W "${flags}" -- ${current_word}) )
+		[[ $COMPREPLY == "--release-stage=" ]] && compopt -o nospace
+	fi
+}
+
+_rmt-cli_repos()
+{
+	local current_word options flags
+
+	current_word=${COMP_WORDS[COMP_CWORD]}
+	options="list enable disable custom help"
+	flags="--all --csv"
+
+	if [[ ${COMP_CWORD} == 2 ]] ; then
+		COMPREPLY=( $(compgen -W "${options}" -- ${current_word}) )
+	elif [[ ${COMP_WORDS[2]} == help && ${COMP_CWORD} == 3 ]] ; then
+		COMPREPLY=( $(compgen -W "${options/help/}" -- ${current_word}) )
+	elif [[ ${COMP_CWORD} > 2 && ${COMP_WORDS[2]} == custom ]] ; then
+		_rmt-cli_repos_custom
+	elif [[ ${current_word} == -* && ${COMP_WORDS[2]} =~ ^(list|ls)$ ]] ; then
+		COMPREPLY=( $(compgen -W "${flags}" -- ${current_word}) )
+	fi
+}
+
+_rmt-cli_repos_custom()
+{
+	local current_word options flags
+
+	current_word=${COMP_WORDS[COMP_CWORD]}
+	options="list add enable disable remove products attach detach help"
+	flags="--csv"
+
+	if [[ ${COMP_CWORD} == 3 ]] ; then
+		COMPREPLY=( $(compgen -W "${options}" -- ${current_word}) )
+	elif [[ ${COMP_WORDS[3]} == help && ${COMP_CWORD} == 4 ]] ; then
+		COMPREPLY=( $(compgen -W "${options/help/}" -- ${current_word}) )
+	elif [[ ${current_word} == -* && ${COMP_WORDS[*]} =~ ^(list|ls|products)$ ]] ; then
+		[[ ${COMP_WORDS[*]} =~ custom ]] && flags="--csv"
+		COMPREPLY=( $(compgen -W "${flags}" -- ${current_word}) )
+	fi
+}
+
+_rmt-cli_import()
+{
+	local current_word options
+
+	current_word=${COMP_WORDS[COMP_CWORD]}
+	options="data repos help"
+
+	if [[ ${COMP_CWORD} == 2 ]] ; then
+		COMPREPLY=( $(compgen -W "${options}" -- ${current_word}) )
+	elif [[ ${COMP_WORDS[2]} == help && ${COMP_CWORD} == 3 ]] ; then
+		COMPREPLY=( $(compgen -W "${options/help/}" -- ${current_word}) )
+	elif [[ ${COMP_CWORD} == 3 ]] ; then
+		COMPREPLY=( $(compgen -f $current_word) )
+	fi
+}
+
+_rmt-cli_export()
+{
+	local current_word options
+
+	current_word=${COMP_WORDS[COMP_CWORD]}
+	options="data settings repos help"
+
+	if [[ ${COMP_CWORD} == 2 ]] ; then
+		COMPREPLY=( $(compgen -W "${options}" -- ${current_word}) )
+	elif [[ ${COMP_WORDS[2]} == help && ${COMP_CWORD} == 3 ]] ; then
+		COMPREPLY=( $(compgen -W "${options/help/}" -- ${current_word}) )
+	elif [[ ${COMP_CWORD} == 3 ]] ; then
+		COMPREPLY=( $(compgen -f ${current_word}) )
+	fi
+}
+
+# alias functions:
```

Comment:
Because I wanted to avoid a "switch case"-like structure, I craft the name of the function to call from the word given on the CLI (in line 26):
```
_rmt-cli_$subcommand
```
And since `repo` and `product` are allowed aliases for `repos` and `products`, I want to also run  completion if somebody typed those words, which forces me to wrap the subcommand completion routines.